### ISSUE TITLE
RFC - mapPropsOnScroll

### ISSUE THREAD

**Issue by evenchange4** [wuct/react-dom-utils#18]
## mapPropsOnScroll
> Attaches the props returned by propsMapper to owner props and updates it when a `scroll` event of the `window` is triggered.

## API
```js
type Scroll = {
  x: number,
  y: number,
};

mapPropsOnScroll(
  propsMapper: (scroll: Scroll, previousScroll: Scroll) => Object,
  throttle: Function,
  BaseComponent: ReactElementType
): ReactElementType
```

## Example
```js
mapPropsOnScroll((scroll, previousScroll) => ({
  isScrollUp: previousScroll.y > scroll.y,
})),
```

**Comment by wuct** [comment]
LGTM

**Comment by wuct** [comment]
Closed by #22
